src/ukify/ukify.py

```diff
@@ -550,50 +550,95 @@ def pe_add_sections(uki: UKI, output: str):
 
     pe.write(output)
 
+def signer_sign(cmd):
+    print('+', shell_join(cmd))
+    subprocess.check_call(cmd)
 
-def make_uki(opts):
-    # kernel payload signing
+def find_sbsign(opts=None):
+    return find_tool('sbsign', opts=opts)
 
-    sbsign_tool = find_tool('sbsign', opts=opts)
-    sbsign_invocation = [
+def sbsign_sign(sbsign_tool, input_f, output_f, opts=None):
+    sign_invocation = [
         sbsign_tool,
         '--key', opts.sb_key,
         '--cert', opts.sb_cert,
+        input_f,
+        '--output', output_f,
     ]
-
     if opts.signing_engine is not None:
-        sbsign_invocation += ['--engine', opts.signing_engine]
+        sign_invocation += ['--engine', opts.signing_engine]
+    signer_sign(sign_invocation)
+
+def find_pesign(opts=None):
+    return find_tool('pesign', opts=opts)
+
+def pesign_sign(pesign_tool, input_f, output_f, opts=None):
+    sign_invocation = [
+        pesign_tool, '-s', '--force',
+        '-n', opts.sb_certdir,
+        '-c', opts.sb_cert_name,
+        '-i', input_f,
+        '-o', output_f,
+    ]
+    signer_sign(sign_invocation)
 
-    sign_kernel = opts.sign_kernel
-    if sign_kernel is None and opts.linux is not None and opts.sb_key:
-        # figure out if we should sign the kernel
-        sbverify_tool = find_tool('sbverify', opts=opts)
+SBVERIFY = {
+    'name': 'sbverify',
+    'option': '--list',
+    'output': 'No signature table present',
+}
 
-        cmd = [
-            sbverify_tool,
-            '--list',
-            opts.linux,
-        ]
+PESIGCHECK = {
+    'name': 'pesign',
+    'option': '-i',
+    'output': 'No signatures found.',
+    'flags': '-S'
+}
 
-        print('+', shell_join(cmd))
-        info = subprocess.check_output(cmd, text=True)
+def verify(tool, opts):
+    verify_tool = find_tool(tool['name'], opts=opts)
+    cmd = [
+        verify_tool,
+        tool['option'],
+        opts.linux,
+    ]
+    if 'flags' in tool:
+        cmd.append(tool['flags'])
+
+    print('+', shell_join(cmd))
+    info = subprocess.check_output(cmd, text=True)
+
+    if tool['output'] in info:
+        return True
+    return False
+
+def make_uki(opts):
+    # kernel payload signing
+
+    sign_tool = None
+    if opts.signtool == 'sbsign':
+        sign_tool = find_sbsign(opts=opts)
+        sign = sbsign_sign
+        verify_tool = SBVERIFY
+    else:
+        sign_tool = find_pesign(opts=opts)
+        sign = pesign_sign
+        verify_tool = PESIGCHECK
+
+    sign_args_present = opts.sb_key or opts.sb_cert_name
 
-        # sbverify has wonderful API
-        if 'No signature table present' in info:
-            sign_kernel = True
+    if sign_tool is None and sign_args_present:
+        raise ValueError('%s, required to sign, is not installed!' % opts.signtool)
```

Comment:
No, no, please don't use %.
`f'{opts.signtool}, required for signing, is not installed'`